tests/Unit/PrestaShopBundle/Utils/BoolParserTest.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Tests\Unit\PrestaShopBundle\Utils;
+
+use PHPUnit\Framework\TestCase;
+use PHPUnit_Framework_AssertionFailedError;
+use PrestaShopBundle\Utils\BoolParser;
+
+class BoolParserTest extends TestCase
```

Comment:
@rokaszygmantas @matks created BoolParser to handle cases like: "0", "false" returned from legacy classes/request handlers instead of boolean false; Tested it with phpunit for your convenience.

Reply:
It's a good idea but can we rather store it in `PrestaShop\PrestaShop\Core\Util` namespace (=> `src/Core/Util` folder) ?

`PrestaShopBundle` should contain all the code required to run PrestaShop as a Symfony application. This means that all the code in it must either serve as an util class to make prestashop domain code usable in Symfony or a class dedicated to app layer concerns, such as rendering twig views, handling sessions ...

The idea behind is to have a Core prestashop code folder which contains all the __logic__ of PrestaShop: Cart Rules, Customer logic, Multistore ... that is being wrapped by PrestaShopBundle to make it a Symfony application.
 In an ideal world, tomorrow if we decide to migrate from Symfony to Laravel, we might be able to create a new folder PrestaShopLaravel where we put all the code required to run PrestaShop as a Laravel application, delete PrestaShopBundle and reuse 100% of PrestaShop core folder.

This is why PrestaShop core folder must remain framework-agnostic.

Reply:
Yep, the folder architecture is explained here https://devdocs.prestashop.com/1.7/development/architecture/understanding-src-folder/

Also maybe one day we will be able to extract the whole `Core` folder and have a dependency library with it So then we could have a Symfony implementation AND a laravel one Or maybe other project will simply use the Core library for their own purpose

Reply:
I see from documentation that PrestaShopBundle/Utils should be move to Core eventually.